src/azul/service/async_manifest_service.py

```diff
@@ -63,40 +78,76 @@ class AsyncManifestService(AbstractService):
+        execution_id = input['object_key'].rpartition('/')[2]
+        execution_id = execution_id.partition('.')[0]
+        token = Token(execution_id=f'{execution_id}_0',
+                      request_index=0,
+                      wait_time=self._get_next_wait_time(0))
+        for retry_index in range(self.step_function_execution_retries):
+            try:
+                self.step_function_helper.start_execution(self.state_machine_name,
+                                                          token.execution_id,
+                                                          execution_input=input)
+            except ClientError as e:
+                if e.response['Error']['Code'] == 'ExecutionAlreadyExists':
+                    output, status = self._inspect_generation(token)
+                    if status not in Status:
+                        raise StateMachineError(status, output)
+                    if status is Status.running:
+                        return token.advance(wait_time=self._get_next_wait_time(token.request_index))
+                    else:
+                        # If a manifest expires before the name of the Step
+                        # Function execution that created it can be reused
+                        # (90 days), StartExecution will fail with
+                        # ExecutionAlreadyExists. In that case we need to retry
+                        # even a successful execution, so that it can recreate
+                        # the manifest.
+                        token = Token(execution_id=f'{execution_id}_{retry_index + 1}',
+                                      request_index=0,
+                                      wait_time=self._get_next_wait_time(0))
+                else:
+                    raise
             else:
-                raise
-        output = execution.get('output', None)
-        status = execution['status']
-        if status == 'SUCCEEDED':
+                return token
+        else:
+            raise StateMachineError('Maximum manifest generation retries reached.',
+                                    token.execution_id)
+
+    def inspect_generation(self, token) -> Union[Token, JSON]:
+        output, status = self._inspect_generation(token)
+        if status is Status.succeeded:
             # Because describe_execution is eventually consistent, output may
             # not yet be present
             if output is None:
                 return token.advance(wait_time=1)
             else:
                 return json.loads(output)
-        elif status == 'RUNNING':
+        elif status is Status.running:
             return token.advance(wait_time=self._get_next_wait_time(token.request_index))
         else:
-            raise StateMachineError(status, output)
+            raise StateMachineError(status.value, output)
+
+    def _inspect_generation(self, token: Token) -> Tuple[str, Status]:
+        try:
+            helper = self.step_function_helper
```

Comment:
Maybe we should just rename the attribute in a first commit.

Reply:
@hannes-ucsc, notice that commit, this fixup! commit is to the first commit which addressed some refactoring and renaming. I swapped this change to the first commit, did you mean for it to have its own commit?

Reply:
Yes.

Reply:
The commit isn't the first commit and it doesn't rename any attributes.